File: ros_basics/src/string_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

#include <iostream>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "string_publisher");
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::String>("/string",10);
	ros::Rate loop_rate(10);

	while(ros::ok())
	{
		std_msgs::String msg;
		msg.data = "Hello World";
		ROS_INFO("Publisher sent: %s", msg.data.c_str());
		pub.publish(msg);
		ros::spinOnce();
		loop_rate.sleep();

	}
	return 0;

}

File: ros_basics/scripts/string_publisher.py
#!/usr/bin/env python
import rospy 
from std_msgs.msg import String

def publisher():
	pub = rospy.Publisher("string", String, queue_size = 10)
	rospy.init_node("string_publisher", anonymous=True)
	rate = rospy.Rate(10)

	while not rospy.is_shutdown():
		str = "hello from Publisher" 
		rospy.loginfo("Publisher sent : %s", str)
		pub.publish(str)
		rate.sleep()

if __name__ == '__main__':
	try:
		publisher()
	except rospy.ROSInterruptException:
		pass